app/uk/gov/hmrc/apisubscriptionfields/repository/FieldsDefinitionRepository.scala

```diff
@@ -72,19 +76,28 @@ class FieldsDefinitionMongoRepository @Inject()(mongoDbProvider: MongoDbProvider
     collection.find(selector).one[FieldsDefinition]
   }
 
-  override def save(fieldsDefinition: FieldsDefinition): Future[Unit] = {
-    val selector = Json.obj("id" -> fieldsDefinition.id)
-    Logger.debug(s"[save] selector: $selector")
-    collection.find(selector).one[BSONDocument].flatMap {
-      case Some(document) => collection.update(selector = BSONDocument("_id" -> document.get("_id")), update = fieldsDefinition)
-      case _ => collection.insert(fieldsDefinition)
-    }.map {
-      writeResult => handleError(writeResult, s"Could not save fields definition fields: $fieldsDefinition")
+  /**
+    * Saves or inserts entity depending on if it already exists.
+    * Returns Future of isInserted Boolean flag if everything went OK - otherwise a failed Future with an error message.
+    * @param fieldsDefinition entity to upsert
+    */
+  def upsert(fieldsDefinition: FieldsDefinition): Future[Boolean] = {
+    collection.update(selector = BSONDocument("_id" -> fieldsDefinition.id), update = fieldsDefinition, upsert = true).map {
```

Comment:
`_id` is the mongo primary key.
`id` is the field we defined.
They are two different things.

I suspect this code might have something wrong.